libraries/bezier-rs/src/bezier/transform.rs

```diff
@@ -237,18 +251,136 @@ impl Bezier {
+
+		match self.handles {
+			BezierHandles::Linear => Bezier::from_linear_dvec2(transformed_start, transformed_end),
+			BezierHandles::Quadratic { handle } => {
+				let handle_scale_distance = (start_distance + end_distance) / 2.;
+				let transformed_handle = utils::scale_point_from_origin(handle, intersection, should_flip_direction, handle_scale_distance);
+				Bezier::from_quadratic_dvec2(transformed_start, transformed_handle, transformed_end)
+			}
+			BezierHandles::Cubic { handle_start, handle_end } => {
+				let handle_start_scale_distance = (start_distance * 2. + end_distance) / 3.;
+				let transformed_handle_start = utils::scale_point_from_origin(handle_start, intersection, should_flip_direction, handle_start_scale_distance);
+
+				let handle_end_scale_distance = (start_distance + end_distance * 2.) / 3.;
+				let transformed_handle_end = utils::scale_point_from_origin(handle_end, intersection, should_flip_direction, handle_end_scale_distance);
+				Bezier::from_cubic_dvec2(transformed_start, transformed_handle_start, transformed_handle_end, transformed_end)
+			}
+		}
+	}
+
 	/// Offset will get all the reduceable subcurves, and for each subcurve, it will scale the subcurve a set distance away from the original curve.
 	/// Note that not all bezier curves are possible to offset, so this function first reduces the curve to scalable segments and then offsets those segments.
 	/// A proof for why this is true can be found in the [Curve offsetting section](https://pomax.github.io/bezierinfo/#offsetting) of Pomax's bezier curve primer.
 	/// Offset takes the following parameter:
-	/// - `distance` - The distance away from the curve that the new one will be offset to. Positive values will offset the curve in the same direction as the endpoint normals,
+	/// - `distance` - The offset's distance from the curve. Positive values will offset the curve in the same direction as the endpoint normals,
 	/// while negative values will offset in the opposite direction.
 	pub fn offset(&self, distance: f64) -> Vec<Bezier> {
 		let mut reduced = self.reduce(None);
 		reduced.iter_mut().for_each(|bezier| *bezier = bezier.scale(distance));
 		reduced
 	}
 
+	/// Version of the `offset` function which scales the offset such that the start of the osset is `start_distance` from the original curve, while the end of
+	/// of the offset is `end_distance` from the original curve. The curve transitions from `start_distance` to `end_distance` gradually, proportional to the
+	/// distance along the equation (`t`-value) of the curve.
+	pub fn graduated_offset(&self, start_distance: f64, end_distance: f64) -> Vec<Bezier> {
+		let reduced = self.reduce(None);
+		let mut next_start_distance = start_distance;
+		let distance_difference = end_distance - start_distance;
+		let total_length = self.length(None);
+
+		let mut result = vec![];
+		reduced.iter().for_each(|bezier| {
```

Comment:
This function seems to be approximating the result, maybe make a comment about that.